src/main/java/taintedmagic/common/items/equipment/ItemVoidwalkerBoots.java

```diff
@@ -157,21 +163,44 @@ public void movementEffects(EntityPlayer player, float bonus, ItemStack itemStac
                 }
                 player.jumpMovementFactor = 0.00002F;
             } else if (Hover.getHover(player.getEntityId())) {
-                player.jumpMovementFactor = 0.03F * speedMod;
+                player.jumpMovementFactor = 0.03F;
 
             } else {
-                player.jumpMovementFactor = 0.05F * speedMod;
+                player.jumpMovementFactor = 0.05F;
             }
         }
     }
 
+    public static float sashBuff(final EntityPlayer player) {
+        // speed & jump buff should only apply when sash+voidwalkers are worn together
+        ItemStack boots = player.getCurrentArmor(0);
+        if (boots != null && boots.getItem() instanceof IVoidwalker) {
+            for (ItemStack stack : PlayerHandler.getPlayerBaubles(player).stackList) {
+                if (stack != null && stack.getItem() == ItemRegistry.ItemVoidwalkerSash
+                        && (stack.stackTagCompound != null && stack.stackTagCompound.getBoolean("mode"))) {
+                    return 0.4F;
+                }
+            }
+        }
+        return 0.0F;
+    }
```

Comment:
It's a style thing, but returning early instead of nesting checks is a lot cleaner.

I also suggested that this could be moved to ItemVoidwalkerSash because it's more relevant for that item, but a better place might be a default method in IVoidwalker. That way, boots could modify how much of a speed boost they get by overriding that method. You could replace the 0.4F with a getSashSpeedIncrease() method that returns 0.4F by default to make it easier to change.
```suggestion
    public static float sashBuff(final EntityPlayer player) {
        // speed & jump buff should only apply when sash+voidwalkers are worn together
        ItemStack boots = player.getCurrentArmor(0);
        if (boots == null || !(boots.getItem() instanceof IVoidwalker)) {
            return 0.0F;
        }
        for (ItemStack stack : PlayerHandler.getPlayerBaubles(player).stackList) {
            if (stack != null && stack.getItem() == ItemRegistry.ItemVoidwalkerSash
                    && (stack.stackTagCompound != null && stack.stackTagCompound.getBoolean("mode"))) {
                return 0.4F;
            }
        }
        return 0.0F;
    }
```

Reply:
Should be the only place across the codebase that has the sash speed, making it a config is probably for a more general code cleanup pr.